docs/openapi/openapi.yaml

```diff
@@ -0,0 +1,118 @@
+openapi: "3.0.2"
+info:
+  title: LPA ID Service
+  version: "1.0"
+  license:
+    name: MIT
+    url: https://opensource.org/licenses/MIT
+x-amazon-apigateway-policy:
+  Version: "2012-10-17"
+  Statement:
+    - Effect: Allow
+      Principal:
+        AWS: ${allowed_roles}
+      Action: "execute-api:Invoke"
+      Resource:
+        - ${lambda_invoke_arn}
+security:
+  - {}
+paths:
+  /cases:
+    post:
+      operationId: createCase
+      summary: Generate a case ID
+      requestBody:
+        content:
+          application/json:
+            schema:
+              type: object
+              required:
+                - type
+                - source
+                - donor
+              properties:
+                type:
+                  type: string
+                  enum:
+                    - "hw"
+                    - "pfa"
+                source:
+                  type: string
+                  enum:
+                    - "APPLICANT"
+                    - "PHONE"
+                donor:
+                  type: object
+                  required:
+                    - name
+                    - dob
+                    - postcode
+                  properties:
+                    name:
+                      type: string
+                      example: Jacalyn Petralba
+                    dob:
+                      type: string
+                      format: date
+                    postcode:
+                      type: string
+                      maxLength: 8
+                      example: EC1A 1BB
```

Comment:
should we strip spaces & set the max to 7 so that we're strict on validation from the start?

Reply:
Yep fair point, updated!